sway/commands/gaps.c

```diff
@@ -0,0 +1,186 @@
+#include <string.h>
+#include "sway/commands.h"
+#include "sway/config.h"
+#include "sway/tree/arrange.h"
+#include "log.h"
+#include "stringop.h"
+#include <math.h>
+
+enum gaps_op {
+	GAPS_OP_SET,
+	GAPS_OP_ADD,
+	GAPS_OP_SUBTRACT
+};
+
+enum gaps_scope {
+	GAPS_SCOPE_ALL,
+	GAPS_SCOPE_WORKSPACE,
+	GAPS_SCOPE_CURRENT
+};
+
+double read_double(char *str) {
+	if (strcmp(str, "0") == 0 || strcmp(str, "0.0") == 0) {
```

Comment:
You could just read errno for ERANGE further on instead of explicitly checking for 0.